sci-mathematics/calc/calc-2.12.5.3.recipe

```diff
@@ -0,0 +1,131 @@
+executes that single command and exits. Otherwise, it enters interactive mode. \
+In this mode, it accepts commands one at a time, processes them, and displays \
+the answers. In the simplest case, commands are simply expressions which are \
+evaluated."
+HOMEPAGE="http://www.isthe.com/chongo/tech/comp/calc/"
+COPYRIGHT="1999-2016 Landon Curt Noll"
+LICENSE="GNU LGPL v2.1"
+REVISION="1"
+SOURCE_URI="http://www.isthe.com/chongo/src/calc/calc-$portVersion.tar.bz2"
+CHECKSUM_SHA256="0fcf60917efb10aab145f1180c278a3ab63d789aad6d5e5f066a4458b5f9b692"
+PATCHES="calc-$portVersion.patchset"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86_gcc2 x86"
+
+PROVIDES="
+	calc$secondaryArchSuffix = $portVersion
+	cmd:calc$secondaryArchSuffix = $portVersion
+	cmd:calc_static$secondaryArchSuffix = $portVersion
+	cmd:cscript$secondaryArchSuffix = $portVersion
+	lib:libcalc$secondaryArchSuffix
+	lib:libcustcalc$secondaryArchSuffix
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libreadline$secondaryArchSuffix
+	lib:libhistory$secondaryArchSuffix
+	lib:libncurses$secondaryArchSuffix
+	calc_data == $portVersion
+	"
+
+if [ -z "$secondaryArchSuffix" ]; then
+	SUMMARY_data="CHANGE ME LATER"
+	ARCHITECTURES_data="any"
+	PROVIDES_data="
+		calc_data = $portVersion
+		"
+	REQUIRES_data=""
+fi
+
+PROVIDES_devel="
+	calc${secondaryArchSuffix}_devel = $portVersion
+	devel:libcalc$secondaryArchSuffix
+	devel:libcustcalc$secondaryArchSuffix
+	"
+REQUIRES_devel="
+	calc$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libreadline$secondaryArchSuffix
+	devel:libhistory$secondaryArchSuffix
+	devel:libncurses$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:awk
+	cmd:sed
+	cmd:diff
+	cmd:egrep
+	cmd:ctags
+	cmd:xargs
+	cmd:cmp
+	cmd:touch
+	cmd:less
+	cmd:strip
+	"
+
+BUILD()
+{
+	make all \
+		BLD_TYPE=calc-static-only \
+		LCC=gcc \
+		LN="ln -s" \
+		LDCONFIG= \
+		INCDIR="$includeDir" \
+		LIBDIR=$libDir \
+		BINDIR=$binDir \
+		SCRIPTDIR=$libDir/calc/cscript \
+		CALC_SHAREDIR=$dataDir/calc \
+		MANDIR=$manDir/man1 \
+	 	USE_READLINE=-DUSE_READLINE \
+		READLINE_LIB=-lreadline \
+		READLINE_EXTRAS="-lhistory -lncurses" \
+		EXTRA_CFLAGS="-DUSE_TERMIOS -DHAVE_UID_T"
+}
+
+INSTALL()
+{
+	make install \
+		BLD_TYPE=calc-static-only \
+		LCC=gcc \
+		LN="ln -s" \
+		LDCONFIG= \
+		INCDIR="$includeDir" \
+		LIBDIR=$libDir \
+		BINDIR=$binDir \
+		SCRIPTDIR=$libDir/calc/cscript \
+		CALC_SHAREDIR=$dataDir/calc \
+		MANDIR=$manDir/man1 \
+	 	USE_READLINE=-DUSE_READLINE \
+		READLINE_LIB=-lreadline \
+		READLINE_EXTRAS="-lhistory -lncurses" \
+		EXTRA_CFLAGS="-DUSE_TERMIOS -DHAVE_UID_T"
+
+	rm $binDir/calc && ln -s calc-static $binDir/calc
+
+	prepareInstalledDevelLibs libcalc libcustcalc
+	fixPkgconfig
+
+	cd $dataDir/calc
+	find -type f -name \*.cal | xargs chmod +w
+	if [ -z "$secondaryArchSuffix" ]; then
+		packageEntries data \
+			$dataDir/calc
+	fi
```

Comment:
Can you please replace this block:
```
	if [ -z "$secondaryArchSuffix" ]; then
		packageEntries data \
			$dataDir/calc
	fi
```
by these two blocks:
```
	if [ -z "$secondaryArchSuffix" ]; then

		# Some files in $dataDir/calc/help/ have code snippets with
		# long shebangs that we can shorten. Currenly, these files are
		# cscript, full, libcalc and usage.
		sed -i -e "1 s|$binDir/calc |/$relativeBinDir/calc |" \
			$dataDir/calc/help/*

		packageEntries data \
			$dataDir/calc
	else
		rm -rf $dataDir/calc $manDir
	fi

	# Shorten the paths to calc in the shebangs of the files in cscript.
	sed -i -e "1 s|$binDir/calc |/$relativeBinDir/calc |" \
		$libDir/calc/cscript/*
```
The changes in the first block will ensure that we don't ship the /system/data/calc/ in the base package of secondary arch builds. It also fixes the shebangs in a few files in /system/data/calc/data/help/.

The second block fixes the shebangs in the scripts that will go to /system/lib/calc/cscripts/ (on primary arch) or  /system/lib/x86/calc/cscripts/ (on x86 secondary arch).